Mathlib/Order/OrdinalApproximantsFixedPoints.lean

```diff
@@ -0,0 +1,365 @@
+  have h₂ := not_le_of_lt (Cardinal.succ_lt #α)
+  exact h₂ h₁
+
+end Cardinal
+
+
+
+namespace Monotone
+
+
+universe u v
+variable {α : Type u}
+variable {β : Type v}
+variable [Preorder α] [PartialOrder β] (f: α → β)
+
+/-- If a monotone function is equal at two points, it is equal between all of them -/
+theorem monotone_stabilizing {a₁ a₂ : α} (h_mon : Monotone f) (h_fa : f a₁ = f a₂):
+    ∀ i, a₂ ≥ i → i ≥ a₁ → f i = f a₁ := by
+  intro i h₂ h₁
+  apply le_antisymm
+  · rw[h_fa]; exact h_mon h₂
+  · exact h_mon h₁
+
+/-- If an antitone function is equal at two points, it is equal between all of them -/
+theorem antitone_stabilizing {a₁ a₂ : α} (h_anti : Antitone f) (h_fa : f a₁ = f a₂):
+    ∀ i, a₂ ≥ i → i ≥ a₁ → f i = f a₁ := by
+  intro i h₂ h₁
+  apply le_antisymm
+  · exact h_anti h₁
+  · rw[h_fa]; exact h_anti h₂
+
+end Monotone
+
+/-We loosly follow the proof from Frederico Enchenique in A Short and Constructive Proof of
+  Tarski's fixed-point theorem published in the International Journal of Game Theory Volume 33(2),
+  June 2005, pages 215-218. -/
+
+namespace OrdinalApprox
+
+universe u
+variable {α : Type u}
+variable [CompleteLattice α] (f : α →o α)
+
+open Function fixedPoints Cardinal Order OrderHom
+
+set_option linter.unusedVariables false in
+/-- Ordinal approximants of the least fixed points -/
+def lfp_approx (a : Ordinal.{u}) : α :=
+  sSup { f (lfp_approx b) | (b : Ordinal) (h : b < a) }
+termination_by lfp_approx a => a
+decreasing_by exact h
+
+theorem lfp_approx_monotone : Monotone (lfp_approx f) := by
+  unfold Monotone; intros a b h; unfold lfp_approx
+  refine sSup_le_sSup ?h; simp
+  intros a' h'
+  use a'; apply And.intro; exact lt_of_lt_of_le h' h
+  exact rfl
+
+/-- Ordinal approximants of the least fixed points as an order homomorphism -/
+def lfp_approx_hom : Ordinal.{u} →o α where
+  toFun i := lfp_approx f i
+  monotone' a b h := by simp; apply lfp_approx_monotone f h
+
+theorem lfp_approx_addition (a : Ordinal.{u}) : f (lfp_approx f a) = lfp_approx f (a+1) := by
+  apply le_antisymm
+  · conv => right; unfold lfp_approx
+    apply le_sSup; simp
+    use a
+  · conv => left; unfold lfp_approx
+    apply sSup_le; simp
+    intros a' h
+    apply f.2; apply lfp_approx_monotone; exact h
+
+/-- The ordinal approximants of the least fixed points are stabilizing
+  when reaching a fixed point of f -/
+theorem lfp_approx_stabilizing_at_fp (a : Ordinal.{u}) (h: lfp_approx f a ∈ (fixedPoints f)):
+    ∀ b > a, lfp_approx f b = lfp_approx f a := by
+  intro b hab; rw[mem_fixedPoints_iff] at h;
+  induction b using Ordinal.induction with | h b IH =>
+  apply le_antisymm
+  · conv => left; unfold lfp_approx
+    apply sSup_le; simp
+    intro a' ha'b
+    by_cases haa : a' < a
+    · rw[lfp_approx_addition]
+      apply lfp_approx_monotone
+      simp; exact haa
+    · simp at haa
+      cases (le_iff_lt_or_eq.mp haa) with
+      | inl haa => specialize IH a' ha'b haa; rw[IH, h];
+      | inr haa => rw[← haa, h];
```

Comment:
There should not be a need to end a line with a semicolon. Please remove them in the entire PR